constraints/recursiveCatEvals.js

```diff
@@ -56,26 +67,53 @@ var sCat = ["cp", "xp", "x0"];
  * 7/29/19 refactor of an earlier version
```

Comment:
Please expand on this comment to explain the definitions of minimal and maximal we're working with when requireLexical and requireOvertHeads come into play.
Also provide a list of options in the comment before the function.